open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +3818,176 @@ components:
+          type: boolean
+          default: true
+        snapshot-id:
+          description:
+            The ID of the snapshot to use for the table scan.
+            If not specified, the snapshot at the main branch head will be used.
+          type: integer
+          format: int64
+        use-snapshot-schema:
+          description:
+            If the schema of the specific snapshot should be used instead of the table schema.
+          type: boolean
+          default: false
+
+    IncrementalSnapshotScanContext:
+      description:
+        Context for scanning data appended in a range of snapshots.
+        The scan always follows the schema of the snapshot at the main branch head.
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/BaseTableScanContext'
+      required:
+        - type
+        - start-snapshot-id
+      properties:
+        type:
+          type: string
+          enum: ["incremental-snapshot-scan"]
+        select:
+          $ref: '#/components/schemas/SelectedFieldNames'
+        filter:
+          $ref: '#/components/schemas/Filter'
+        case-sensitive:
+          description: If field selection and filtering should be case sensitive
+          type: boolean
+          default: true
+        start-snapshot-id:
+          description: The ID of the starting snapshot of the incremental scan
+          type: integer
+          format: int64
+        inclusive-start:
+          description: If the data appended in the start snapshot should be included in the scan
+          type: boolean
+          default: false
+        end-snapshot-id:
+          description:
+            The ID of the inclusive ending snapshot of the incremental scan.
+            If not specified, the snapshot at the main branch head will be used as the end snapshot.
+          type: integer
+          format: int64
+
+    FieldName:
+      description:
+        A field name that follows the Iceberg naming standard, and can be used in APIs like
+        Java `Schema#findField(String name)`.
+
+        The nested field name follows these rules
+        - nested struct fields are named by concatenating field names at each struct level using dot (`.`) delimiter,
+        e.g. employer.contact_info.address.zip_code
+        - nested fields in a map key are named using the keyword `key`, e.g. employee_address_map.key.first_name
+        - nested fields in a map value are named using the keyword `value`, e.g. employee_address_map.value.zip_code
+        - nested fields in a list are named using the keyword `element`, e.g. employees.element.first_name
+      type: string
+
+    SelectedFieldNames:
+      description:
+        A list of fields in schema that are selected in a table scan.
+        When not specified, all columns in the requested schema should be selected.
+      type: array
+      items:
+        $ref: '#/components/schemas/FieldName'
+
+    Filter:
```

Comment:
I don't see a need for this type. If the filter is not specified, then the service should not apply a filter rather than sending a default value. While it may amount to the same thing, I think it is cleaner not to have extra types like this one.

Reply:
@rdblue 
For services if we do not give any clear indication on how to interpret an empty filter expression (in this case default to true), how should they handle this? I think this type gives a clear guideline, so there is some standard on what clients can expect in terms of results. 

cc @jackye1995 @amogh-jahagirdar

Reply:
I can also remove this type `Filter` and just map it back to Expression, but I think we need to be clear with the default value.

Reply:
@rahil-c Since the residual filter will just be an expression and that expression is optional https://github.com/apache/iceberg/pull/9695/files#r1689067372, I actually don't think we really need to define a default value. The benefit to having it be optional is that clients can compute residuals on their own as an optimization since server side computation of residuals may be too expensive. 

So in brief:

If a server sends back a residual -> client should use that for filtering data
If a server doesn't send back a residual -> client can compute the residual on it's own or apply the original filter.

Then considering this we really don't need the extra type since it's really just an expression at this point.

I think that's what @rdblue was getting at and I think I agree with it.

Reply:
There were a couple of uses of `Filter`, but we already have an expression type and I don't see a reason to make a subtype that has a default. It is up to the request that uses an expression whether an expression is optional and what the semantics are if an expression is not included.

When using an expression in the scan context's `filter`, then if there _is not filter_ I think the behavior is fairly clear (do not filter). We can explicitly state that as well if it anyone thinks that it is needed, but I don't think that we need to require that clients send `"filter": {"type": "true"}`.

Reply:
With the current structure of the API, this is no longer a problem.